.github/workflows/build-test-windows-x86.jsonnet

```diff
@@ -55,6 +55,15 @@ default: https://github.com/ocaml/opam-repository.git
         make PREFIX="$prefix" install
       |||,
     },
+    {
+      name: 'Cache opam switch',
```

Comment:
I just made some helpers in libs/semgrep.libsonnet called cache_opam. Maybe you can use it.
Also can you explain restore-keys, I don't understand.

Reply:
Also setup-ocaml itself is using some cache, so would be good to explain why we need another cache, what are the differences with the setup-ocaml cache.

Reply:
Added some comments. I don't think I can use `cache_opam` as-is, but maybe we can generalize it a bit in a subsequent PR -- I'd like to get this merged so that any additional windows improvements I make benefit from the build speedup (shaves 7-8 mins off build time)